app/models/nic/interface.rb

```diff
@@ -105,12 +105,12 @@ def normalize_name
       # no hostname was given or a domain was selected, since this is before validation we need to ignore
       # it and let the validations to produce an error
       return if name.empty?
-      if domain.nil? && name.include?('.') && changed_attributes['domain_id'].blank?
+      if Domain.unscoped.find_by(:id => domain_id).nil? && name.include?('.') && changed_attributes['domain_id'].blank?
```

Comment:
I think this can be `if domain_id.nil?...`, since we are checking if the nic has a domain set so no need for the additional lookup. tbh this function could benefit from some refactoring, but not required in the scope of this pr